service/books_search/book_search_handler.go

```diff
@@ -0,0 +1,37 @@
+package books_search
+
+import (
+	"fmt"
+	"gin/config"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+func validateSearchRequest(req *bookSearchRequest) error {
+	if req.Title == "" && req.Name == "" && !(req.containsPriceFilter()) {
+		return fmt.Errorf("request must contain at least one filter")
+	}
+
+	return nil
+}
+
+func SearchBook(c *gin.Context) {
+	req, err := searchRequestBuilder(c)
```

Comment:
Maybe just parse `title` `authorName` and `priceRange` and pass them on to the ES `searchbooks` function.
I understand the need here for a struct but in the other routes I want to avoid a "request struct" because there is no need, so let's try not to use one here as well. 
WDYT?

Also, let's avoid using the maxInt. If given a one-sided range let's create a one-sided range in ES.

Reply:
I disagree, parsing 3 (and in the future, maybe more) parameters and passing them everywhere is cumbersome.
I thing it is better to encapsulate them into structs when there is more then a single parameter.

Reply:
NP you can capsule them.